rfcs/len-metamethod-rawlen.md

```diff
@@ -0,0 +1,40 @@
+# Support `__len` metamethod for tables and `rawlen` function
+
+## Summary
+
+`__len` metamethod will be called by `#` operator on tables, matching Lua 5.2
+
+## Motivation
+
+Lua 5.1 invokes `__len` only on userdata objects, whereas Lua 5.2 extends this to tables. In addition to making `__len` metamethod more uniform and making Luau
+more compatible with later versions of Lua, this has the important advantage which is that it makes it possible to implement an index based container.
+
+Before `__iter` and `__len` it was possible to implement a custom container using `__index`/`__newindex`, but to iterate through the container a custom function was
+necessary, because Luau didn't support generalized iteration, `__pairs`/`__ipairs` from Lua 5.2, or `#` override.
+
+With generalized iteration, a custom container can implement its own iteration behavior so as long as code uses `for k,v in obj` iteration style, the container can
+be interfaced with the same way as a table. However, when the container uses integer indices, manual iteration via `#` would still not work - which is required for some
+more complicated algorithms, or even to simply iterate through the container backwards.
+
+Supporting `__len` would make it possible to implement a custom integer based container that exposes the same interface as a table does.
+
+## Design
+
+`#v` will call `__len` metamethod if the object is a table and the metamethod exists; the result of the metamethod will be returned.
```

Comment:
We also want the result to always be a `number`.

Reply:
Yup.